include/zephyr/net_buf.h

```diff
@@ -1832,6 +1833,72 @@ static inline void net_buf_add_be64(struct net_buf *buf, uint64_t val)
 	net_buf_simple_add_be64(&buf->b, val);
 }
 
+/**
+ * @brief Add a character string to the end of the buffer
+ *
+ * Adds a @c \\0 terminated string to buffer if it fits.
+ * The terminating @c \\0 itself is not added.
+ *
+ * See @ref net_buf_simple_to_str to access the data as a regular C string.
+ *
+ * @param buf Buffer to update.
+ * @param s String to be added.
+ *
+ * @return 0 if the formatted string could fit.
+ * @return -ENOMEM if the buffer was too short for the formatted string.
+ */
+int net_buf_simple_add_str(struct net_buf_simple *buf, const char *s);
+
+/**
+ * @brief Append a formatted string to the buffer if it fits
+ *
+ * Like @ref net_buf_simple_sprintf but called with a @c va_list instead of a
+ * variable number of arguments.
+ *
+ * @param buf Buffer to update.
+ * @param fmt Format string as used with snprintf().
+ * @param ... Additional arguments formatted by @c fmt
+ */
+int net_buf_simple_vsprintf(struct net_buf_simple *buf, const char *fmt, va_list ap);
+
+/**
+ * @brief Append a formatted string to the buffer if it fits,
+ *
+ * Adds a formatted string to the end of a buffer, like the @c snprintf()
+ * function does, only if the resulting does not fit, or leave the buffer
+ * unchanged otherwise.
+ *
+ * See @ref net_buf_simple_to_str to access the data as a regular C string.
+ *
+ * @param buf Buffer to update.
+ * @param fmt Format string as used with snprintf().
+ * @param ... Additional arguments formatted by @c fmt
+ *
+ * @return 0 if the formatted string could fit
+ * @return -ENOMEM if the buffer was too short for the formatted string.
+ */
+static inline int net_buf_simple_sprintf(struct net_buf_simple *buf, const char *fmt, ...)
+{
+	va_list ap;
+
+	va_start(ap, fmt);
+	return net_buf_simple_vsprintf(buf, fmt, ap);
+}
+
+/**
+ * @brief Get a C string representation of the buffer with a trailing @c \\ 0.
+ *
+ * Ensure the buffer has a @c \\0 terminator to the C string, and if there was
+ * room for it return the pointer to the terminated buffer.
+ *
+ * @param buf Buffer to access as a C string.
+ *
+ * @return A pointer to the buffer as a string, or NULL if there was no room for
+ *         the @c \\0 terminator. The string returned can still be modified
+ *         through @c buf. Exclusive access must be ensured by the caller.
+ */
+char *net_buf_simple_to_str(struct net_buf_simple *buf);
```

Comment:
I don't quite understand the use case of this. How for example zero byte would be handled?

Reply:
This is a word of warning related to this:

> no guarantee that `str` stays safe to use, as `buf` may still be modified

For instance:

```c
/* This adds `\0` to the buffer data, but the `\0` is not part of the buffer itself */
str = net_buf_simple_to_str(buf);
if (str == NULL) {
	return -ENOMEM;
}

data->name = str;

if (net_buf_tailroom(buf) < 2) {
	return -ENOMEM;
}

/* This will remove the `\0` terminator, and `str` will further have the wrong content */
net_buf_add_mem(buf, "XY", 2);

return 0;
```

This would not be a problem if instead of `data->name = str;` it was `if (strcmp("str", key) == 0) {...}`: temporarily use string.h functions in the middle of a net\_buf flow.

This is one way I thought. Maybe there are better ones? What would be best to access the net_buf buffers as strings?